src/execution_plan/optimizations/apply_join.c

```diff
@@ -189,7 +189,6 @@ static void _reduce_cp_to_hashjoin(ExecutionPlan *plan, OpBase *cp) {
 					_re_order_filter_op(plan, cp, additional_filter);
```

Comment:
Line 182
`for(int k = j + 1; k < filter_count; k++) {`
I thought we agreed to perform this loop only once, outside of "main while loop: `while(filter_count > 0) {`